### DISCUSSION TITLE
Make video player suitable to video playing direction

### DISCUSSION THREAD

**Discussion by Edfgdz** [Docile-Alligator/Infinity-For-Reddit#1020] [Ideas]
Is it possible that make video rotate base on if de vice is vertical or horizontal? 

For example when people put phone vertically then all video will be played in portrait mode and when people put phone horizontally then video would also be rotated to landscape mode. 

Advantage of this method is that people could rotate video more easily based on if video is landscape or portrait. 

So wish developer could consider it.

**Comment by Edfgdz** [comment]
I reconsider this problem and have another thought about this problem. 

At present about video playing problem is that some videos in reddit should be played in portrait mode such as this post 

https://www.reddit.com/r/Infinity_For_Reddit/comments/pisp87/posted_from_the_official_reddit_app_can_access/

So if it is possible to let Infinity app to detect if video is portrait or landscape and make Infinity app video player automatically change to suitable playing style?

For example if video is portrait then player changed to portrait and if video landscape then to landscape too.

**Comment by Wladefant** [comment]
@Edfgdz do you mean that you want a toggle in the settings where the mode of the phone (landscape, portrait) is not important for the video rotation but the suitability of the video?

**Reply by Edfgdz** [reply]
Yeah,that is what i mean,like this setting in mxplayer 

![IMG_20220821_181519](https://user-images.githubusercontent.com/26460124/185786367-4a5c7d9b-41be-47e0-8092-ed92a6fe5468.png)

**Comment by obar** [comment]
I think I have the same feature proposal, let me know if it differs.

In a nutshell, the idea is rather than having a toggle to "switch to landscape orientation in video player automatically," it could be more useful to have that setting to be "automatically rotate video player to fit content".

That would mean, after [the test for that to-be-renamed setting](https://github.com/Docile-Alligator/Infinity-For-Reddit/blob/8a2932122ce839c60c8370fbd029cdebceddd16b/app/src/main/java/ml/docilealligator/infinityforreddit/activities/ViewVideoActivity.java#L369) the code would still read the current orientation, then (in pseudocode)

```
boolean changeOrientation = false;
Integer targetOrientation = currentOrientation;
if (currentOrientation == PORTRAIT) {
  if (video.width > video.height) {
    changeOrientation = true;
    targetOrientation = LANDSCAPE;
  }
} else {
  if (video.height > video.width) {
    changeOrientation = true;
    targetOrientation = PORTRAIT;
  }
}

if (changeOrientation) {
  setRequestedOrientation(targetOrientation);
  ...
}
```

This accounts for square videos, which might as well be played in the current orientation.
